### ISSUE TITLE
[qwen3] Loss is extremely hight when initialized with seed checkpoint

### ISSUE THREAD

**Issue by wwwjn** [pytorch/torchtitan#1879]
### Bug description

The seed checkpoint was initialized on CPU with following command, and the high loss is reproducible with the seed checkpoint created in following command:
```
NGPU=1 CONFIG_FILE="./torchtitan/models/qwen3/train_configs/qwen3_0.6b.toml" ./run_train.sh --checkpoint.enable --checkpoint.create_seed_checkpoint --parallelism.data_parallel_replicate_degree 1 --parallelism.data_parallel_shard_degree 1 --parallelism.tensor_parallel_degree 1 --parallelism.pipeline_parallel_degree 1 --parallelism.context_parallel_degree 1 --parallelism.expert_parallel_degree 1
```

With the initialized checkpoint, run job with FSDP=2 and TP=2, the loss are extremely high

<img width="3094" height="670" alt="Image" src="https://github.com/user-attachments/assets/503768fa-cc3a-4ee4-be67-d89b96359a6e" /> 

### Versions

torchtitan main branch [6bccdb6](https://github.com/pytorch/torchtitan/commit/6bccdb652847e6fdcc259092d4b057146cd343ef)

**Comment by rakkit** [comment]
emm. I just noticed there is no `model.safetensors.index.json` file in 0.6b's [repo](https://huggingface.co/Qwen/Qwen3-0.6B), so torchtitan will not read the HF checkpoint at all.   

but the problem also appears for 1.7b model

**Comment by tianyu-l** [comment]
@rakkit 
> so torchtitan will not read the HF checkpoint at all.

Are you sure? @ankitageorge told me if there's no `.index.json`, everything (the only file) will be read.

**Comment by ankitageorge** [comment]
Yes, at least from the DCP side, we can read checkpoints without index.json file without any issues

> [@rakkit](https://github.com/rakkit)
> 
> > so torchtitan will not read the HF checkpoint at all.
> 
> Are you sure? [@ankitageorge](https://github.com/ankitageorge) told me if there's no `.index.json`, everything (the only file) will be read.

**Comment by rakkit** [comment]
@wwwjn noob question.  i was checking qwen's code in `transformers`. both [qwen3](https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen3/modeling_qwen3.py#L158) and [qwen3moe](https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen3_moe/modeling_qwen3_moe.py#L63) use the same `apply_rotary_pos_emb` with `split-halves` layout `[x0, x1,...xn; y0, y1,....yn]` same as [llama(2,3)](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L145). So is there any reason we dont permute/unperpute in qwen3's `state_dict_adapter.py` ? for DCP<->HF convert.

**Comment by wwwjn** [comment]
> `transformers`. both [qwen3](https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen3/modeling_qwen3.py#L158) and [qwen3moe](https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen3_moe/modeling_qwen3_moe.py#L63) use the same `apply_rotary_pos_emb` with `split-halves` layout `[x0, x1,...xn; y0, y1,....yn]` same as [llama(2,3)](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L145). So is there any reason we dont permute/unperpute in qwen3's `state_d`

This related how the weights layout and RoPE implementation are published on HF.  Each weight layout published on HF needs to align with the RoPE implementation. 

For llama model, the published weight on HF aligns with their RoPE implementation (their RoPE implementation assume the `xq` is `split-half` layout) , so torchtitan need to permute back because our RoPE is different.
- More about our RoPE implementation: our RoPE implementation assumes`xq` is interleaved-layout [x0, y0, x1, y1, ...], where (x0, y0) represents the first complex number. 

For Qwen3 model, our RoPE implementation is the same as transformer's.

**Comment by wwwjn** [comment]
Verified after #2342  landed, the training is much more stable: 

<img width="1427" height="472" alt="Image" src="https://github.com/user-attachments/assets/83ddd89e-9486-4645-b439-d6787c5e23f8" />

The problem is because the std of output layer initialized is too high if we just reused the tok_embedding layer initial weights. Weight tying means output.weight and tok_embeddings.weight share the same tensor. But they have conflicting initialization needs:

  - Embedding (tok_embeddings): typically initialized with nn.init.normal_(weight) (std=1.0)
  - Output (output): needs small std like dim**-0.5 (e.g., ~0.014 for dim=4096)
